addons/grenadethrowing/functions/fnc_isFFVAndWeaponUp.sqf

```diff
@@ -0,0 +1,27 @@
+/*
+ * Author: Dslyecxi, Jonpas
+ * Checks if in firing from vehicles seat and has a weapon up.
+ *
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * Is FFV and has weapon up <BOOL>
+ *
+ * Example:
+ * [] call ace_grenadethrowing_fnc_isFFVAndWeaponUp
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+private _return = false;
+
+if (vehicle player == player) exitWith {true};
+
+private _assignedRole = assignedVehicleRole player;
```

Comment:
`assignedVehicleRole` returns what your AI group leader want's you to be and not our actual vehicle position.